src/components/Tree/expandCollapseButton.js

```diff
@@ -11,16 +12,27 @@ function getIcon(isExpanded) {
     return <RightArrow />;
 }
 
+function getButton(isExpanded, isLoading, onClick) {
+    const spinner = <BrandSpinner variant="brand" size="x-small" />;
+    return !isExpanded && isLoading ? (
+        <Button size="x-small" variant="base" icon={spinner} disabled />
+    ) : (
+        <Button size="x-small" icon={getIcon(isExpanded)} onClick={onClick} />
+    );
+}
```

Comment:
please use regular if condition, it's a better reading

Reply:
also we should put an spinner when is loading, not a button with spinner inside

Reply:
received, working on it

Reply:
solved according to the comments